docs/blog/2019-07-26-how-i-used-theme-ui-to-build-my-gatsby-themes-library/index.md

```diff
@@ -0,0 +1,176 @@
+
+```yaml
+shapes: [
+  {
+    type: "circle",
+    color: "green",
+    size: ["200px", "300px"],
+    xOffset: ["-140px", "-120px"],
+    yOffset: ["-70px"],
+    opacity: 0.5,
+  },
+  {
+    type: "donut",
+    color: "teal",
+    size: ["25px", "100px"],
+    xOffset: ["50px"],
+    yOffset: ["-60px"],
+    opacity: 1,
+  },
+]
+```
+
+These are all the information that my shape components need to render the desired output. I didn't use _left_ and _right_ here as the shapes should always face the outside (on the inside it could overlap the text). As you can see the array / breakpoint notation for position and size was used — I think that's just cool 😎
+
+To give you an idea how I then added this functionality to the listing here's an excerpt of the relevant file:
+
+```js:title=src/components/listing.tsx
+<Container sx={{ py: 4 }}>
+  {themes.map((theme, index) => {
+    const isEven = index % 2 === 0
+
+    return (
+      <Box>
+        <div>
+          {theme.shapes.map(shape => {
+            switch (shape.type) {
+              case `circle`:
+                return (
+                  <Circle
+                    key={`${shape.size}-${shape.color}-${shape.xOffset}`}
+                    size={shape.size}
+                    color={shape.color}
+                    left={isEven ? shape.xOffset : `unset`}
+                    right={isEven ? `unset` : shape.xOffset}
+                    top={shape.yOffset}
+                    sx={{ zIndex: 0, opacity: shape.opacity }}
+                  />
+                )
+              case `donut`:
+                return (
+                  <Donut
+                    key={`${shape.size}-${shape.color}-${shape.xOffset}`}
+                    width="20px"
+                    size={shape.size}
+                    color={shape.color}
+                    left={isEven ? shape.xOffset : `unset`}
+                    right={isEven ? `unset` : shape.xOffset}
+                    top={shape.yOffset}
+                    sx={{ zIndex: 0, opacity: shape.opacity }}
+                  />
+                )
+              default:
+                return null
+            }
+          })}
+        </div>
+      </Box>
+    )
+  })}
+</Container>
+```
+
+Setting the offset values to either _left_ or _right_ is decided by _isEven_ which is also used to change the order of the items. Passing in the arrays into the props works like on the rest of the site and hence the responsive styling!
+
+## Conclusion
+
+All in all Theme UI helped me a lot to get this done quickly. Starting at the Design to Code Handoff (by using the same design tokens) and ending at using the breakpoint syntax in the GraphQL layer of Gatsby to individually style theme entries. Of course Theme UI offers you all the things it says on its own website: Ergonomic, Themeable, Contraint-based. And in the case of Gatsby Themes: Also shadowable.
```

Comment:
```suggestion
All in all Theme UI helped me a lot to get this done quickly. From the design phase, to the code handoff phase (by using the same design tokens) to finally using the breakpoint syntax in the GraphQL layer of Gatsby to individually style theme entries. Of course Theme UI offers you all the things it says on its own website: "Ergonomic, Themeable, Contraint-based". And in the case of Gatsby themes: Also shadowable.
```

Reply:
> All in all Theme UI helped me a lot to get this done quickly. Starting at the Design to Code Handoff (by using the same design tokens) and ending at using the breakpoint syntax in the GraphQL layer of Gatsby to individually style theme entries.

I'm not _entirely_ sure I read this sentence correctly, does my suggestion make sense?

Reply:
Yeah, I realize now that it works like that in german but not english 😅 It should summarize the process in a short manner and I think your suggestion also does that